quiffen/core/qif.py

```diff
@@ -191,6 +191,17 @@ def parse(cls, filepath, separator='\n', day_first=True):
                                                'present')
                 accounts[default_account.name] = default_account
                 last_account = default_account.name
+
+                # if there is a transaction in this section, parse it (often seen at head of qifs)
+
+                if len(section.split('\n')) > 2: # !Type:... and ^ 
```

Comment:
Is it only `>2` in the case of a transaction, or are there potentially other things that can appear here?

Reply:
I think on reflection I'd want to do something a bit different here. 

The issue is that if the first section has a header line and transaction details in it, then a default account is created without processing the transaction details. Per the qif documentation on wikipedia (I know...) there is no caret between the header and the first transaction. So what we should probably do is:

* swap the order of the `elif '!Type:Invst' in header_line:` and `elif '!Type' in header_line and not accounts:` tests
* amend `elif '!Type' in header_line and not accounts:` to `elif not accounts and header_line in VALID_TRANSACTION_ACCOUNT_TYPES:`
* after creating the default account, send the first section to the transaction processor without testing for >2 lines

The investment type already creates the transaction for the first section, it is just the other `VALID_TRANSACTION_ACCOUNT_TYPES` that need the same treatment, assuming Wiki is accurate

Reply:
I think this falls apart in the case where there _is_ a caret between the account definition and the transaction though, does it not?

Reply:
I added a test case for the circumstance where the file looks like 

```
!Type:Bank
^
D01-12-2000
T1.00
```
and it failed. I added logic which will detect when we are in a header-row-only section (which will have length two when split by `\n` due to the residue from originally splitting by `^`), and the parser now won't attempt to create a transaction when there aren't enough rows for this to happen. 

I think this kind of file is invalid per the spec but maybe there are some like it out there.